python/treelite/gtil/gtil.py

```diff
@@ -2,16 +2,50 @@
 General Tree Inference Library (GTIL)
 """
 import ctypes
+import json
+import warnings
+from typing import Optional
+
 import numpy as np
-from ..frontend import Model
+
 from ..core import _LIB, _check_call
+from ..frontend import Model
+from ..util import c_str
+
 
+class GTILConfig:
+    """Object holding configuration data"""
 
-def predict(model: Model, data: np.ndarray, nthread: int = -1, pred_margin: bool = False):
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, *, nthread: int, predict_type: str):
+        predictor_config = {
+            "nthread": nthread,
+            "predict_type": predict_type,
+        }
+        predictor_config = json.dumps(predictor_config)
+        self.handle = ctypes.c_void_p()
+        _check_call(
+            _LIB.TreeliteGTILParseConfig(
+                c_str(predictor_config), ctypes.byref(self.handle)
+            )
+        )
+
+    def __del__(self):
+        if self.handle is not None:
+            _check_call(_LIB.TreeliteGTILDeleteConfig(self.handle))
+
+
+def predict(
+    model: Model,
+    data: np.ndarray,
+    *,
+    nthread: int = -1,
+    pred_margin: Optional[bool] = None,
+    predict_type: Optional[str] = None
```

Comment:
I would suggest to call this "return_type" or so to make clear that this actually changes the nature of the return value.

Reply:
On a second thought, I actually prefer having separate functions: `predict_leaf`, `predict_per_tree`. Let me factor them out into separate functions.